src/Microsoft.DotNet.MSBuildSdkResolver/Interop.NETStandard.cs

```diff
@@ -16,6 +17,16 @@ internal static partial class Interop
     {
         internal static readonly bool RunningOnWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
 
+        internal static string realpath (string path)
+        {
+            var buffer = new StringBuilder(4096); // PATH_MAX from linux/limits.h
```

Comment:
> http://man7.org/linux/man-pages/man3/realpath.3.html

> BUGS      
>
>       The POSIX.1-2001 standard version of this function is broken by
>       design, since it is impossible to determine a suitable size for the
>       output buffer, resolved_path.  According to POSIX.1-2001 a buffer of
>       size PATH_MAX suffices, but PATH_MAX need not be a defined constant,
>       and may have to be obtained using pathconf(3).  And asking
>       pathconf(3) does not really help, since, on the one hand POSIX warns
>       that the result of pathconf(3) may be huge and unsuitable for
>       mallocing memory, and on the other hand pathconf(3) may return -1 to
>       signify that PATH_MAX is not bounded.  The resolved_path == NULL
>       feature, not standardized in POSIX.1-2001, but standardized in
>       POSIX.1-2008, allows this design problem to be avoided.

I would think we can rely on the POSIX.1-2008 feature here (pass NULL, `free()` result when done). It is documented as supported in the FreeBSD and Mac OS X man pages as well.